### DISCUSSION TITLE
adding new filterstore

### DISCUSSION THREAD

**Discussion by solganik** [seaweedfs/seaweedfs#2625] [Ideas]
Hi @chrislusf  I wanted to add foundationdb as a filerstore as it seems quite sutable for the schenario, it is fast on reads and allows efficient prefix searches.
https://github.com/solganik/seaweedfs/commit/931d9f0cea615cc8be45d99579739774c14ab44e (WIP)
How can i test filerstore? are there any functional / benchmark tests for it ?

**Comment by chrislusf** [comment]
Thanks! This will be great!

Yes, need some tests for this. Now is a good time to add one. :)

**Comment by solganik** [comment]
Ok implementation done and i had tested it with manual fofoo tests .. I dont upload it to PR yet as it is not ready (no tests)
It would be really nice if you have time to take a look at https://github.com/solganik/seaweedfs/tree/foundationdb
Also i have a question .. it this possible that filer entries are large ? (i.e. entries themselves in the filer) ? can they get above 100Kbytes ?

**Reply by chrislusf** [reply]
you can create a PR for line by line review.

If one file entry has too many file chunks, the file chunks will be merged into a manifest file chunk.

Do you want to add transactions support for foundationdb?

**Comment by solganik** [comment]
Thats also a question i wanted to ask. This is why i need to know how big
values can be. Fdb has limitation on 5 sec transaction. And max value
length of 100k. If values is larger it needs to split.
So can value exceed 100k? How do i simulate this?

**Reply by chrislusf** [reply]
If so, this would not work if renaming a large folder. We can go without transaction.

For value more than 100KB, you can try to create a file with more than 2000 chunks. Assuming chunk size is 4MB, the file needs to be more than 8GB. You can adjust the chunk size to be smaller.

**Comment by solganik** [comment]
what i can do for transaction is keeping all in memory (all objects) and only set them on commit..
i`ll work on compilation docker for fdb as as awesome that project is .. it is hard to install :) 
i`ll check with chunks .. it means that i`ll have to re-write the subfoder prefix search .. but still it wont suffer from "large subfolder problem" as FDB has legixcografic order on keys and iterator ..
